src/lib/es2020.intl.d.ts

```diff
@@ -262,28 +198,154 @@ declare namespace Intl {
     };
 
     interface NumberFormatOptions {
-        compactDisplay?: string;
-        notation?: string;
-        signDisplay?: string;
-        unit?: string;
-        unitDisplay?: string;
+        compactDisplay?: "short" | "long" | undefined;
+        notation?: "standard" | "scientific" | "engineering" | "compact" | undefined;
+        signDisplay?: "auto" | "never" | "always" | undefined;
+        unit?: NumberFormatUnit | undefined;
+        unitDisplay?: "short" | "long" | "narrow" | undefined;
     }
 
     interface ResolvedNumberFormatOptions {
-        compactDisplay?: string;
-        notation?: string;
-        signDisplay?: string;
-        unit?: string;
-        unitDisplay?: string;
+        compactDisplay?: "short" | "long";
+        notation?: "standard" | "scientific" | "engineering" | "compact";
+        signDisplay?: "auto" | "never" | "always";
+        unit?: NumberFormatUnit;
+        unitDisplay?: "short" | "long" | "narrow";
+    }
+
+    interface DateTimeFormatOptions {
+        calendar?: string | undefined;
+        dayPeriod?: "narrow" | "short" | "long" | undefined;
+        numberingSystem?: string | undefined;
+        hourCycle?: "h11" | "h12" | "h23" | "h24" | undefined;
     }
 
     interface DateTimeFormatOptions {
-        dateStyle?: "full" | "long" | "medium" | "short";
-        timeStyle?: "full" | "long" | "medium" | "short";
+        dateStyle?: "full" | "long" | "medium" | "short" | undefined;
+        timeStyle?: "full" | "long" | "medium" | "short" | undefined;
+        hourCycle?: "h11" | "h12" | "h23" | "h24" | undefined;
+    }
+
+    type LocaleHourCycleKey = "h12" | "h23" | "h11" | "h24";
+    type LocaleCollationCaseFirst = "upper" | "lower" | "false";
+
+    interface LocaleOptions {
+        /** A string containing the language, and the script and region if available. */
+        baseName?: string;

```

Comment:
shouldn't all these optional properties add `| undefined`?

Reply:
I added the `| undefined` to the user inputs, but not the returned objects from the API - so _I think_ that leaves this as OK but will re-check